src/main/java/com/hillel/items_exchange/service/AdvertisementService.java

```diff
@@ -54,6 +55,11 @@ public AdvertisementDto createAdvertisement(AdvertisementDto advertisementDto, U
         adv.setStatus(Status.NEW);
         adv.getProduct().setSubcategory(subcategoryService.findById(advertisementDto.getProduct().getSubcategoryId())
                 .orElseThrow(EntityNotFoundException::new));
+
+        Optional<Location> location = locationService.findByCityAndDistrict(
+                adv.getLocation().getCity(), adv.getLocation().getDistrict());
+        adv.setLocation(location.orElseGet(() -> locationService.createLocation(adv.getLocation())));
+
```

Comment:
I remember that you have discussed the logic of adding Location if it does not exist, but what about the case when we, for example, receive the city "Kharkiv" in the first advertisement and city "Kharkov" in the second (for now we create new Location). I think we need to discuss adding a new Location one more time in the next meeting.